src/cli/managers/config_manager.py

```diff
@@ -21,9 +47,29 @@ def get_applications(self) -> List[Dict[str, str]]:
         return [{}]
 
     def application_exists(self, application: str) -> bool:
-        return True
+        """
+        Checks if the application name already exists in .qne/application.json for unique purposes. If the file is
+        non-existing, the application won't exist and returns True. If it does exist, return False, else True.
+
+        Args:
+            application: the application name
+
+        """
+
+        if self.__check_and_create_config():
+            return False
+
+        app_config_file = Path.home() / ".qne/applications.json"
+
+        data = utils.read_json_file(app_config_file)
+
+        for key in data:
+            if key == application:
```

Comment:
But if I check if the application is unique using the path + application_name and I create the same application_name in another directory it will succeed but the application will be overwritten in the json file (because it the key value will be the application_name).

Reply:
I meant for the entry appname of course. So first get the path for the appname, but the application really exists (and was not deleted from the file system by the user) when the directory (with the appname) exists.
I added QNEBE-281 to fix this.